src/Libraries/DSOfficeUtilities/Excel.cs

```diff
@@ -0,0 +1,533 @@
+                // just swallow this exception and try to create an new excel
+                // instance.
+
+                // Sometimes a FileNotFoundException occurs with DynamoCore.XmlSerializers
+                // which is not clear why. This exception also makes Excel tests very flaky
+                // It is found that just swallowing these exceptions and creating a new excel instance 
+                // below causes the test to continue and successfuly pass
+            }
+
+            if (excel == null)
+            {
+                excel = new Microsoft.Office.Interop.Excel.Application();
+            }
+
+            excel.Visible = ShowOnStartup;
+            excel.DisplayAlerts = false;
+
+            return excel;
+        }
+
+        /// <summary>
+        /// Check if the excel process is running
+        /// </summary>
+        public static bool ExcelProcessRunning
+        {
+            get
+            {
+                return Process.GetProcessesByName("EXCEL").Length != 0;
+            }
+        }
+
+        /// <summary>
+        /// Check if this object holds a reference to Excel
+        /// </summary>
+        public static bool HasValidExcelReference
+        {
+            get { return _app != null; }
+        }
+
+        /// <summary>
+        /// Close all Excel workbooks and provide SaveAs dialog if needed.  Also, perform
+        /// garbage collection and remove references to Excel App
+        /// </summary>
+        private static void TryQuitAndCleanup(bool saveWorkbooks)
+        {
+            if (HasValidExcelReference)
+            {
+                if (ExcelProcessRunning)
+                {
+                    App.Workbooks.Cast<Workbook>().ToList().ForEach((wb) => wb.Close(saveWorkbooks));
+                    App.Quit();
+                }
+
+                while (Marshal.ReleaseComObject(_app) > 0)
+                {
+
+                }
+
+                _app = null;
+            }
+        }
+
+        internal static void OnProcessExit(object sender, EventArgs eventArgs)
+        {
+            if (eventArgs != null)
+            {
+                var args = eventArgs as ExcelCloseEventArgs;
+                if (args != null)
+                {
+                    TryQuitAndCleanup(args.SaveWorkbooks);
+                }
+            }
+            else
+                TryQuitAndCleanup(true);
+        }
+    }
+
+    internal static class ExcelWriteUtils
+    {
+        internal static object[][] WriteData(string filePath, string sheetName, int startRow, int startCol, object[][] data, bool overWrite = false, bool writeAsString = false)
+        {
+            WorkBook wb = new WorkBook(filePath);
+            WorkSheet ws = new WorkSheet(wb, sheetName, overWrite);
+            ws = ws.WriteData(startRow, startCol, data, writeAsString);
+            return ws.Data;
+        }
+    }
+
+    [IsVisibleInDynamoLibrary(false)]
```

Comment:
I think we can remove these attributes as this project is no longer a node library.